app/services/story_query_builder.rb

```diff
@@ -50,6 +57,41 @@ def published_filter
     end
   end
 
+  def state_filter
+    if params[:state] === 'published'
+      Filter.new do
+        range published_at: { lte: 'now' }
+      end
+    elsif params[:state] === 'unpublished'
+      Filter.new do
```

Comment:
And here is where the actual ES filters are built.  Range filters are pretty simple ... but for this "unpublished" one I needed to combine (with a `bool`) the must-not-exist OR range.